### PR TITLE
Update: NSIS.NSIS version 3.07

### PR DESCRIPTION
**Pull request by sredna**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/24060)

### CODE REVIEW

File: manifests/n/NSIS/NSIS/3.07/NSIS.NSIS.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+FileExtensions: [nsi, nsh]
+
+Scope: machine
+InstallModes: [interactive, silent]
+MinimumOSVersion: 4.0.1381.0
+Installers:
+- Architecture: x86
+  InstallerUrl: https://sourceforge.net/projects/nsis/files/NSIS%203/3.07/nsis-3.07-setup.exe/download
+  InstallerSha256: 82dbb5c52755e46477039bc9c4b13831b03fdcfd6f54a58b59b91802145383cf
+  InstallerType: nullsoft
```
Comment (vedantmgoyal2009) [inline manifests/n/NSIS/NSIS/3.07/NSIS.NSIS.yaml:20]:
```suggestion
Moniker: nsis
Tags: 
- setup
- installer
FileExtensions: 
- nsi
- nsh
InstallModes: 
- interactive
- silent
MinimumOSVersion: 4.0.1381.0
Installers:
- Architecture: x86
  InstallerUrl: https://sourceforge.net/projects/nsis/files/NSIS%203/3.07/nsis-3.07-setup.exe/download
  InstallerSha256: 82dbb5c52755e46477039bc9c4b13831b03fdcfd6f54a58b59b91802145383cf
  InstallerType: nullsoft
  Scope: machine
```

**Review by vedantmgoyal9 — CHANGES_REQUESTED** [review]

File: manifests/n/NSIS/NSIS/3.07/NSIS.NSIS.yaml (hunk 1 above)
Comment (sredna) [reply manifests/n/NSIS/NSIS/3.07/NSIS.NSIS.yaml:20]: Why? [ ] is valid YAML syntax and passes the validator

`Scope:` is allowed in the root and all installers will support the same scope.

Comment (vedantmgoyal2009) [reply manifests/n/NSIS/NSIS/3.07/NSIS.NSIS.yaml:20]: The manifests was absolutely correct, but to

Comment (vedantmgoyal2009) [reply manifests/n/NSIS/NSIS/3.07/NSIS.NSIS.yaml:20]: The manifests were absolutely correct. Just to make it look like other manifests, I suggested changes.

**Review by OfficialEsco — APPROVED** [review]

**Review by OfficialEsco — APPROVED** [review]

**Pull request merged**